lib/src/kbasesearchengine/main/NarrativeInfoDecorator.java

```diff
@@ -81,12 +85,19 @@ public SearchTypesOutput searchTypes(final SearchTypesInput params, final String
     public SearchObjectsOutput searchObjects(final SearchObjectsInput params, final String user)
             throws Exception {
         SearchObjectsOutput searchObjsOutput = searchInterface.searchObjects(params, user);
+        if (searchObjsOutput.getAdditionalProperties().get(removedGuids) == null &&
+                "true".equals(System.getProperty(removedGuidsEnv))){
```

Comment:
From what I understand, they both work for this case. Since javadoc recommended I use system properties, I went with that. Also there were other parts of the code that used the system properties, I figured we wanted to keep it consistent.

Reply:
Can you verify that environment variables show up in the system properties? I didn't think that was the case.

> Since javadoc recommended I use system properties, I went with that.

What javadoc are you talking about?

Reply:
https://docs.oracle.com/javase/7/docs/api/java/lang/System.html#EnvironmentVSSystemProperties
I don't think env variables show up in system properties? Also suppose if we were to use environment variables, how would you set/unset the env variable in a test?

Reply:
So if we do want to use env vars then we need changes here. I don't have strong feelings but if we don't use env vars then there needs to be more engineering to set the sys props on startup. Env vars would be easier to deal with.

https://github.com/kbase/KBaseSearchEngine/blob/master/test/src/kbasesearchengine/test/common/TestCommon.java#L266

Reply:
That link for changing env vars seems super hacky. What if we do something where, on startup, we have a check where if the env variable is set, then setup the System properties? And then the tests can do whatever with the system properties.

Reply:
> That link for changing env vars seems super hacky.

It is, but it's just for tests, so who cares.

> What if we do something where, on startup, we have a check where if the env variable is set, then setup the System properties?

Why not just check the environment instead of the sys props?

Reply:
The hack is used for setting up the environment in tests so the API can be started.

I'm not sure what you mean about local KBaseSearch. To run the the server, you have to set the deployment config env var appropriately.

Reply:
For example https://github.com/kbase/KBaseSearchEngine/blob/master/Dockerfile.release#L17

Reply:
It's okay. Since I don't really have a preference, i can just do it this way